File: benchmarks/oopsla25/subjects/066.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

const char* f066(const char *s) {
    if (strlen(s) > 3) {
        fprintf(stderr, "Exception\n");
        exit(1);
    } else {
        static char result[4];
        strncpy(result, s, 3);
        result[3] = '\0';
        return result;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f066(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/066.py
#!/usr/bin/env python3

import sys

def f066(s: str):
  if len(s) > 3:
    raise Exception
  else:
    return s[0:3]

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f066(my_string))
